eth/internal/api.go

```diff
@@ -0,0 +1,258 @@
+package internal
```

Comment:
Typically, I like keeping APIs as small as possible. In this case, I think it's best to leave this as-is to minimize the diff between the copied version and our version. I'll disable linting for the package as well to avoid modifying the style.